### PR TITLE
TMA-711: segments filter in users brick

### PR DESCRIPTION
**Pull request by kubamahnert**

### CODE REVIEW

File: lib/gooddata/lcm/actions/synchronize_users.rb [hunk 1]
```diff
@@ -168,11 +168,19 @@ def call(params)
                       end
                     when 'sync_domain_client_workspaces'
                       domain_clients = domain.clients(:all, data_product)
+                      if params.segments_filter
+                        segment_uris = params.segments_filter.map(&:uri)
+                        domain_clients.select! { |c| segment_uris.include?(c.segment_uri) }
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: Calls 'c.segment_uri' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: Calls 'segment_uris.include?(c.segment_uri)' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: Has the variable name 'c' (UncommunicativeVariableName)

File: lib/gooddata/lcm/actions/synchronize_users.rb [hunk 2]
```diff
@@ -168,11 +168,19 @@ def call(params)
                       end
                     when 'sync_domain_client_workspaces'
                       domain_clients = domain.clients(:all, data_product)
+                      if params.segments_filter
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: Calls 'params.segments_filter' 4 times (DuplicateMethodCall)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/581/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/377/ : SUCCESS in 33s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/681/ : SUCCESS in 33s

File: spec/unit/actions/synchronize_user_filters_spec.rb [hunk 3]
```diff
@@ -55,18 +55,26 @@
...
           allow(File).to receive(:open).and_return("client_id\n123456789")
           allow(project).to receive(:deleted?).and_return false
+          allow(organization_not_in_segment).to receive(:segment_uri).and_return('another-segment-uri')
         end
         it 'returns results' do
           result = subject.class.call(params)
           expect(result).to eq(results: [[{}]])
         end
+
+        it 'only uses clients from segments in segments filter' do
```
Comment (yenkins) [inline spec/unit/actions/synchronize_user_filters_spec.rb]: Identical code found in synchronize_user_filters_spec.rb:73, synchronize_users_spec.rb:107 (mass = 72)

File: spec/unit/actions/synchronize_users_spec.rb [hunk 4]
```diff
@@ -72,5 +76,39 @@
...
+        allow(domain).to receive(:clients).with(:all, nil).and_return([organization, organization_not_in_segment])
+        allow(domain).to receive(:clients).with('123456789', nil).and_return(organization)
+        allow(segment).to receive(:uri).and_return(segment_uri)
+        allow(organization).to receive(:segment_uri).and_return(segment_uri)
+        allow(organization_not_in_segment).to receive(:segment_uri).and_return('another-segment-uri')
+        allow(organization).to receive(:project).and_return(project)
+        allow(organization).to receive(:client_id).and_return('123456789')
+      end
+
+      it 'only uses clients from segments in segments filter' do
```
Comment (yenkins) [inline spec/unit/actions/synchronize_users_spec.rb]: Identical code found in synchronize_user_filters_spec.rb:73, synchronize_users_spec.rb:107 (mass = 72)

File: lib/gooddata/lcm/actions/synchronize_user_filters.rb [hunk 5]
```diff
@@ -166,18 +166,17 @@ def call(params)
             end
 
             domain_clients = domain.clients(:all, data_product)
-            params.segments_filter ||= params.segments_filter
             if params.segments_filter
-              segments_filter = params.segments_filter.map { |seg| "/gdc/domains/#{domain.name}/segments/#{seg}" }
-              domain_clients.select! { |c| segments_filter.include?(c.segment_uri) }
+              segment_uris = params.segments_filter.map(&:uri)
+              domain_clients.select! { |c| segment_uris.include?(c.segment_uri) }
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_user_filters.rb]: Calls 'c.segment_uri' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_user_filters.rb]: Calls 'segment_uris.include?(c.segment_uri)' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_user_filters.rb]: Has the variable name 'c' (UncommunicativeVariableName)

File: lib/gooddata/lcm/actions/synchronize_user_filters.rb [hunk 6]
```diff
@@ -166,18 +166,17 @@ def call(params)
             end
 
             domain_clients = domain.clients(:all, data_product)
-            params.segments_filter ||= params.segments_filter
             if params.segments_filter
-              segments_filter = params.segments_filter.map { |seg| "/gdc/domains/#{domain.name}/segments/#{seg}" }
-              domain_clients.select! { |c| segments_filter.include?(c.segment_uri) }
+              segment_uris = params.segments_filter.map(&:uri)
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_user_filters.rb]: Calls 'params.segments_filter' 4 times (DuplicateMethodCall)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/582/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/378/ : FAILURE in 1m 47s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/682/ : SUCCESS in 1m 49s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/583/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/380/ : SUCCESS in 33s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/684/ : SUCCESS in 2m 25s

**Comment by panjan** [comment]
ok to merge

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/237/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/381/ : SUCCESS in 30s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/685/ : SUCCESS in 33s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1136/ : SUCCESS in 46m 43s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/894/ : SUCCESS in 27m 00s

**Pull request merged**